crates/arrow_util/src/pretty.rs

```diff
@@ -519,6 +519,13 @@ fn process_batch(
     batch: &RecordBatch,
     rows: Range<usize>,
 ) -> Result<(), ArrowError> {
+    if rows.is_empty() {
+        return Ok(());
+    }
+
+    // Avoid trying to process the entire batch.
+    let batch = batch.slice(rows.start, rows.len());
```

Comment:
what if after the reslice the batch is empty?

Reply:
Then `ColumnValues` will be empty, and nothing will be added to the table since the `for batch_idx in 0..batch.num_rows()` won't have any iterations.